src/Features/Core/Portable/EditAndContinue/AbstractEditAndContinueAnalyzer.cs

```diff
@@ -2367,8 +2367,16 @@ protected static bool TypesEquivalent(ITypeSymbol? oldType, ITypeSymbol? newType
         protected static bool TypesEquivalent<T>(ImmutableArray<T> oldTypes, ImmutableArray<T> newTypes, bool exact) where T : ITypeSymbol
             => oldTypes.SequenceEqual(newTypes, exact, (x, y, exact) => TypesEquivalent(x, y, exact));
 
-        protected static bool ParameterTypesEquivalent(IParameterSymbol oldParameter, IParameterSymbol newParameter, bool exact)
-            => (exact ? s_exactSymbolEqualityComparer : s_runtimeSymbolEqualityComparer).ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
+        protected static bool ParameterTypesEquivalent(IParameterSymbol oldParameter, IParameterSymbol newParameter, bool exact, bool allowDiscards)
+        {
+            if (exact)
+            {
+                return s_exactSymbolEqualityComparer.ParameterEquivalenceComparer.Equals(oldParameter, newParameter);
```

Comment:
>Equals [](http://example.com/codeflow?start=82&length=6)

Shouldn't we compare the names when exact == true?

Reply:
You tell me, you wrote this bit 😛

This method just compares parameter types. If we make it compare names too, we'd just need to do more work later to find out _why_ the parameters were different (to know which rude edit to report, or capability to check).

The discard checking is here because they are more like type changes than name changes, because the type info for them is different.

Reply:
I was a bit confused, but I still don't understand why discards are even considered in this method. Yes, this method should only compare types - ParametersEquivalent should be renamed to ParameterTypesEquivalent (overload). Discard parameter has a type - if it's a method parameter then it has to be explicitly specified. If it is lambda parameter then it may be inferred. 

Discards should imo be considered when we are comparing names. We need to compare names so that we know if we need to emit semantic edit or report diagnostics. That should be separate.

Reply:
> Discard parameter has a type

That's what I was expecting but anecdotally from looking in the debugger it seems that isn't the case. Perhaps @jcouv can confirm, but in https://github.com/dotnet/roslyn/pull/54856#discussion_r678826925 there is a screenshot of the parameter symbol for a discard (on the right) and it has no type information.

Maybe there is something extra that I need to do to get the type info for a discard?

Reply:
I'm trying to dig into this, and if I look at some code in Syntax Visualizer I can see you're correct, but when debugging a test I don't get type info for discards. Trying to work out why though...